test/services/contract_manager.js

```diff
@@ -8,48 +8,97 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
 */
 
 import chai from 'chai';
-import sinon from 'sinon';
-import sinonChai from 'sinon-chai';
+import chaiAsPromised from 'chai-as-promised';
 import ContractManager from '../../src/services/contract_manager';
-import * as Web3Tools from '../../src/utils/web3_tools';
-import resetHistory from '../helpers/reset_history';
+import {createWeb3, deployContract, getDefaultAddress} from '../../src/utils/web3_tools';
+import HeadJson from '../../contracts/Head.json';
+import ContextJson from '../../contracts/Context.json';
 
-chai.use(sinonChai);
+chai.use(chaiAsPromised);
 const {expect} = chai;
 
 describe('Contract Manager', () => {
-  let mockWeb3;
-  let loadContractStub;
-  let deployContractStub;
-  const exampleContract = 'a contract';
-
+  let web3;
+  let head;
+  let context;
+  const atlasStakeStoreAddress = '0x9ee9c302eddcd480e0a87fbcd24d0667f474d957';
+  const bundleStoreAddress = '0x7516eb6cea797169636f5ab1b39386ba50a1a565';
+  const kycWhitelistAddress = '0x3e09d3e466f6b780d60abf88e416643c72d44eb0';
+  const rolesAddress = '0x1cb076e7c1589f0142f2565a2138f0ad14750ad0';
+  const shelteringAddress = '0xd5c9a9960a52fce0e3ea3cba8f505782711d5ad4';
+  const feesAddress = '0xfd70b6abd2f10d5ef83595c4557bce2b4e0fefb8';
+  const challengesAddress = '0x9ae4c0eb11e136f19d46bbe841f76c7a92558dca';
+  const payoutsStoreAddress = '0x5973cdbed51f2c9a34ad99ca34836ed48a157856';
+  const payoutsAddress = '0x09eba34850b5f66f189bc24bee8e519717cea24d';
+  const shelteringTransfersAddress = '0x35cb69c740ff2befafe560d1ab72d8c399cea520';
+  const timeAddress = '0x1c26d2dfdace03f0f6d0aaca233d00728b9e58da';
+  const configAddress = '0x945a3706189472a6855d87b26e9f96ae41268dff';
+  const uploadsAddress = '0xc7b547108934b046becac6e207a6c65f97cc5aa8';
+  const rolesStoreAddress = '0xfa0f2ed3d3ac2905d2b79d25dcf352cf0db8c8d1';
+  const apolloDepositStoreAddress = '0x4e607c633f95739f5ea9c1552b7d44792895ac6b';
+  const addresses = [
+    timeAddress,
+    atlasStakeStoreAddress,
+    bundleStoreAddress,
+    kycWhitelistAddress,
+    rolesAddress,
+    shelteringAddress,
+    feesAddress,
+    challengesAddress,
+    payoutsStoreAddress,
+    payoutsAddress,
+    shelteringTransfersAddress,
+    configAddress,
+    uploadsAddress,
+    rolesStoreAddress,
+    apolloDepositStoreAddress
+  ];
```

Comment:
I think this is extremely fragile and prone to errors as the addresses must be provided in the right order for the context to work. This is somewhat also a integration test, as it uses real Web3 contracts, with real implementations. 

Instead we should mock the web3_utils methods to return sinon.mocks instead of the actual contracts. This way we can unit test this, and check if the contract manager properly uses the contract API, and not test the contracts itself.

On a later stage we should consider wrapping all the web3 methods into an class, so that they can be injected to, making the mocking a lot more clean.

Reply:
done 👍 
(more or less 😝)